rest-client/src/main/java/io/apicurio/registry/client/exception/RestClientException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.client.exception;
+
+import io.apicurio.registry.rest.beans.Error;
+
+/**
+ * @author Carles Arnal <carles.arnal@redhat.com>
+ */
+public class RestClientException extends RuntimeException {
+
+    private final Error error;
+
+    public RestClientException(Error error) {
+        super(error.getMessage());
```

Comment:
Is there any way to parse the error detail into a stack trace that we could then use as a root cause?  Probably not, but worth investigating??

Reply:
Let me take a look.

Reply:
I haven't found a way to convert the String to a stack trace.